### PR TITLE
pass -- as an argument from CLI to Electron application to avoid tripping up Electron check on Windows

### LINKED ISSUE #2518: cypress run -b chrome fails if you attempt to change the baseUrl
### Current behavior:
cypress run fails if you attempt to change the baseUrl from the command line while using the `--browser chrome` option

Running each of these commands individually work as expected with no errors
```> cypress run --browser chrome``` 
```> cypress run --config baseUrl=http://localhost:3000``` 

but strangely, combining them both causes the process to immediately exit with an error.
The failure is almost immediate, there is no output from cypress and no tests are run.
```
> cypress run --browser chrome --config baseUrl=http://localhost:3000

Error: my-project@0.0.0 cypress:cl: `cypress run --browser chrome --config baseUrl=http://localhost:3000`
Exit status 4294967295
    at EventEmitter.<anonymous> (C:\ProgramData\nvm\v8.11.3\node_modules\npm\node_modules\npm-lifecycle\index.js:285:16)
    at emitTwo (events.js:126:13)
    at EventEmitter.emit (events.js:214:7)
    at ChildProcess.<anonymous> (C:\ProgramData\nvm\v8.11.3\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
```

It seemed strange, so I tried combining --browser chrome with other options.
This ran as expected with no errors:
```> cypress run --browser chrome --config chromeWebSecurity=false```

But if you try to set the baseUrl with --env, the same error occurs!
```
> cypress run --browser chrome --env baseUrl=http://localhost:3000
```

My current workaround is to set the CYPRESS_BASE_URL environment variable manually prior to calling cypress run.

### Desired behavior:

```> cypress run --browser chrome --config baseUrl=http://localhost:3000```
should not fail

### Steps to reproduce:

set baseUrl in cypress.json to `localhost:4200`
then run 
```> cypress run --browser chrome --config baseUrl=http://localhost:3000```

### Versions
Cypress 3.1.0
Windows 10 using git bash
Node 8.11.3
Chrome Version 69.0.3497.100

### LINKED ISSUE #5466: cypress shows error in Windows when passing args containing a url in 3.5.0
### Current behavior:

After updating to Cypress 3.5.0 I'm not able to open the Cypress GUI nor to run cypress run anymore. I'm using the Vue-Plugin "cli-plugin-e2e-cypress". I receiving the following error-message:

```ERROR  Error: Command failed: C:\Users\localuser\Desktop\TEST\node_modules\cypress\bin\cypress run --config baseUrl=http://localhost:8082/
    Error: Command failed: C:\Users\localuser\Desktop\TEST\node_modules\cypress\bin\cypress run --config baseUrl=http://localhost:8082/
    at makeError (C:\Users\localuser\Desktop\TEST\node_modules\execa\index.js:174:9)
    at Promise.all.then.arr (C:\Users\localuser\Desktop\TEST\node_modules\execa\index.js:278:16)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
    error Command failed with exit code 1.
    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
 ```

### Desired behavior:

I should be able to run my test with "cypress run" or open the GUI with "cypress open"

### Steps to reproduce: (app code and test code)

- upgrade from cypress 3.4.1 to 3.5.0
- try "cypress run"

### Versions

- Cypress 3.5.0 (cli-plugin-e2e-cypress)
- Windows 10
- Chrome 77

### PR DESCRIPTION
**Pull request by bahmutov**
- Closes #5466
- Fixes #2518

### User facing changelog

Users on Windows should no longer experience a crash trying to pass an url through `--config ...` or through `--env ...` command line arguments when running Cypress tests

### Additional details

Seems Electron application does an initial validation pass over CLI arguments, and if there is anything that looks like an url it will cause a crash. But if we start our arguments with `--` then Electron skips that check, assuming these are app's own flags. So I have added `--` from the CLI and am slicing it out if found in the server.

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue been tagged with a release in ZenHub? 
- [x] added failing test in branch 3.8.0 https://github.com/cypress-io/cypress-example-todomvc/pulls that is tested on Windows

### CODE REVIEW

**Comment by bahmutov** [comment]
Ohh, we still need to pass Electron's own CLI arguments like `--no-sandbox` before `--`

Screen Shot 2019-12-02 at 7 20 14 PM

**Review by flotwig — APPROVED** [review]
@bahmutov have you tested the latest commit manually in Windows and confirmed that it fixes the issue?

If so, LGTM. I wish Electron had this documented somewhere, I can't find it on their site :/

File: cli/lib/exec/spawn.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ module.exports = {
         const electronArgs = _.clone(args)
         const node11WindowsFix = isPlatform('win32')
 
-        if (verify.needsSandbox()) {
+        if (!options.dev && verify.needsSandbox()) {
```
Comment (brian-mann) [inline cli/lib/exec/spawn.js]: why is this necessary? in dev mode we want to be as closely aligned to how it works in production as possible.

Comment (flotwig) [reply cli/lib/exec/spawn.js]: i'm not sure it's necessary, but it works because in `dev` mode the cli launches `node`, not the `Cypress` binary directly

Comment (bahmutov) [reply cli/lib/exec/spawn.js]: if you call Node with this option it blows up
```
$ node --no-sandbox
node: bad option: --no-sandbox
~/git/cypress-test-tiny on master*
$ echo $?
9
```

**Comment by bahmutov** [comment]
From user's comment https://github.com/cypress-io/cypress/issues/5466#issuecomment-561093205 

- recommendation to separate Electron switches from your command line arguments https://electronjs.org/blog/protocol-handler-fix
- closes showing how to pass args https://github.com/electron/electron/pull/13039/files#diff-6428a7ecb6a3a2831eae23aee7efeac5R705

**Pull request merged**